interface/GoodParticle.h

```diff
@@ -0,0 +1,28 @@
+#include <TLorentzVector.h>
+
+// Wrapper for lepton objects so to ignore problems with
+// accessing data if particle is muon vs electron
+
+enum class pType {Muon, Electron};
+
+struct GoodPart {
+  TLorentzVector v;
+  pType ptype;
+  int charge;
+  
+  void SetTVector(double pt, double eta, double phi, double m ) {
+    v.SetPtEtaPhiM(pt, eta, phi, m);
+  }
+
+  void SetpType(pType type) {
+    ptype = type;
+  }
+
+  void SetCharge(int c) {
+    charge = c;
```

Comment:
You could just replace ptype and charge together by pdgId (where the sign tells you if its charge)

Reply:
that's slick....